### PR TITLE
[RemoveDIs] Do not load into new debug info format from bitcode by default

### PR DESCRIPTION
**Pull request by OCHyams**
This is NFC right now, as the global default behaviour is also "do not load into the new debug info format by default", but we want to change that soon.

Additionally unconditionally convert from the new debug info format into if we've loaded into it (e.g., if the bitcode file loaded was already in the new format).

The latter change is needed because verify-uselistorder doesn't yet understand DbgRecords (it doesn't know how to map them).

The former change is needed because if we load from an old debug format bitcode file but load directly into the new format _and then convert back to the old mode after_, the use-lists of the debug intrinsic functions (the functions' global value uses) change.

### CODE REVIEW

**Review by jmorse — COMMENTED** [review]
Why the extra 'r' first in `DbgInforFormat`? Presumably  the intention behind this so that we can widen the scope of RemoveDIs while keeping it turned off for verify-uselistorder?

**Comment by OCHyams** [comment]
> `Why the extra 'r' first in DbgInforFormat?`.

Oops, a typo in the patch that added the flag. I'll fix that separately.

> Presumably the intention behind this so that we can widen the scope of RemoveDIs while keeping it turned off for verify-uselistorder?

Yep, I'm going to upload more patches that enable the bitcode loading by default in specific tools. The intention is that once those stick for a while we can make it the global default.

File: llvm/tools/verify-uselistorder/verify-uselistorder.cpp [hunk 1]
```diff
@@ -68,6 +68,7 @@ static cl::opt<unsigned>
                 cl::desc("Number of times to shuffle and verify use-lists"),
                 cl::init(1), cl::cat(Cat));
 
+extern cl::opt<cl::boolOrDefault> LoadBitcodeIntoNewDbgInforFormat;
```
Comment (jryans) [inline llvm/tools/verify-uselistorder/verify-uselistorder.cpp:71]: Nit: Probably better to keep a blank line after this

**Review by jryans — APPROVED** [review]
Looks reasonable overall following discussion in the PR, thanks!

**Comment by mstorsjo** [comment]
This is causing failed builds here:
```
ld.lld: error: undefined symbol: LoadBitcodeIntoNewDbgInforFormat
>>> referenced by verify-uselistorder.cpp
>>>               tools/verify-uselistorder/CMakeFiles/verify-uselistorder.dir/verify-uselistorder.cpp.o:(main)
>>> did you mean: LoadBitcodeIntoNewDbgInfoFormat
>>> defined in: lib/libLLVMBitReader.a(BitcodeReader.cpp.o)
collect2: error: ld returned 1 exit status
```

I pushed a fix in 336bdf1a255571f8d894e8befe4be7e9141f7541.

**Comment by OCHyams** [comment]
Thanks very much for sorting that out - I renamed the variable (to fix the typo), but didn't remember to update this PR before pushing.

**Pull request merged**